packages/codecs-core/src/fixCodec.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable sort-keys-fix/sort-keys-fix */
+import { fixBytes } from './bytes';
+import { Codec } from './codec';
+import { NotEnoughBytesError } from './errors';
+
+/**
+ * Creates a fixed-size codec from a given codec.
+ *
+ * @param codec - The codec to wrap into a fixed-size codec.
+ * @param fixedBytes - The fixed number of bytes to read.
+ * @param description - A custom description for the codec.
+ */
+export function fixCodec<T, U extends T = T>(
+    codec: Codec<T, U>,
+    fixedBytes: number,
+    description?: string
+): Codec<T, U> {
+    return {
+        description: description ?? `fixed(${fixedBytes}, ${codec.description})`,
+        fixedSize: fixedBytes,
+        maxSize: fixedBytes,
+        encode: (value: T) => fixBytes(codec.encode(value), fixedBytes),
+        decode: (buffer: Uint8Array, offset = 0) => {
```

Comment:
Is it worth scrubbing the word ‘buffer’ from this codebase, since we're taking a stand against `Buffer` in favour of `Uint8Array`?

Reply:
Agreed! I've been renaming them to `bytes` instead but must have missed a few. I'll do a proper search and replace.

Reply:
Done in #1630.